File: CS/Form1.cs
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;

namespace dxExample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            gridView1.Columns["Status"].GroupFormat.FormatType = DevExpress.Utils.FormatType.Numeric;
            gridView1.Columns["Status"].GroupFormat.FormatString = "n3";
        }
        private void Form1_Load(object sender, EventArgs e) {
            InitGridControl();
        }
        private void InitGridControl() {
            DataHelper.InitData(dataTable1);
        }
        private void gridView1_CustomDrawGroupRow(object sender, RowObjectCustomDrawEventArgs e) {
            DrawGroupRow(e);
        }
        private void DrawGroupRow(RowObjectCustomDrawEventArgs e) {
            var info = e.Info as GridGroupRowInfo;
            if (!(info.Column.FieldName == "ID")) {
                return;
            }
            GroupRowPaintHelper.CustomDrawSubjectColumnGroupRow(e, gridView1, gridControl1.LookAndFeel, imageList1);
            e.Handled = true;
        }
        private void gridView1_CustomColumnDisplayText(object sender, CustomColumnDisplayTextEventArgs e) {
            if(e.Column.FieldName == "Subject")
                e.DisplayText = GetCustomColumnDisplayText(e.Value.ToString(), e.ListSourceRowIndex);
        }
        private string GetCustomColumnDisplayText(string groupValueText, int imgIndex) {
            string imgName;
            try {
                imgName = imageCollection1.Images.InnerImages[imgIndex].Name;
            } catch {
                imgName = imageCollection1.Images.InnerImages[0].Name;
            }
            var groupText = string.Format("<image={0}> {1}", imgName, groupValueText);
            return groupText;
        }
    }
}

File: VB/Form1.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms

Namespace dxExample
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
		  '  
			InitializeComponent()
			gridView1.Columns("Status").GroupFormat.FormatType = DevExpress.Utils.FormatType.Numeric
			gridView1.Columns("Status").GroupFormat.FormatString = "n3"
		End Sub
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			InitGridControl()
		End Sub
		Private Sub InitGridControl()
			DataHelper.InitData(dataTable1)
		End Sub
		Private Sub gridView1_CustomDrawGroupRow(ByVal sender As Object, ByVal e As RowObjectCustomDrawEventArgs) Handles gridView1.CustomDrawGroupRow
		   DrawGroupRow(e)
		End Sub
		Private Sub DrawGroupRow(ByVal e As RowObjectCustomDrawEventArgs)
			Dim info = TryCast(e.Info, GridGroupRowInfo)
			If Not(info.Column.FieldName = "ID") Then
				Return
			End If
		   GroupRowPaintHelper.CustomDrawSubjectColumnGroupRow(e, gridView1, gridControl1.LookAndFeel, imageList1)
			e.Handled = True
		End Sub
		Private Sub gridView1_CustomColumnDisplayText(ByVal sender As Object, ByVal e As CustomColumnDisplayTextEventArgs) Handles gridView1.CustomColumnDisplayText
			If e.Column.FieldName = "Subject" Then
				e.DisplayText = GetCustomColumnDisplayText(e.Value.ToString(), e.ListSourceRowIndex)
			End If
		End Sub
		Private Function GetCustomColumnDisplayText(ByVal groupValueText As String, ByVal imgIndex As Integer) As String
			Dim imgName As String
			Try
				imgName = imageCollection1.Images.InnerImages(imgIndex).Name
			Catch
				imgName = imageCollection1.Images.InnerImages(0).Name
			End Try
			Dim groupText = String.Format("<image={0}> {1}", imgName, groupValueText)
			Return groupText
		End Function
	End Class
End Namespace
